opcpa_tpr_config/xpm_prog.py

```diff
@@ -0,0 +1,209 @@
+    instrset.append(FixedRateSync(marker="910kH", occ=1))
+    b5 = len(instrset)
+    instrset.append(ControlRequest([0]))
+    instrset.append(FixedRateSync(marker="910kH", occ=1))
+    instrset.append(Branch.conditional(line=b5, counter=0, value=26))
+    b6 = len(instrset)
+    instrset.append(ControlRequest([0, 1]))
+    instrset.append(FixedRateSync(marker="910kH", occ=1))
+    b7 = len(instrset)
+    instrset.append(ControlRequest([0]))
+    instrset.append(FixedRateSync(marker="910kH", occ=1))
+    instrset.append(Branch.conditional(line=b7, counter=0, value=26))
+    instrset.append(Branch.conditional(line=b6, counter=1, value=323))
+    instrset.append(Branch.conditional(line=b4, counter=2, value=18))
+    instrset.append(Branch.unconditional(line=b0))
+
+    return instrset
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "base_rate",
+        help="Desired laser output rep rate (total)"
+    )
+    parser.add_argument(
+        "goose_rate",
+        help="Desired laser goose rate (sub-harmonic of base_rate)")
+    parser.add_argument("offset", help="Desired 910 kHz bucket offset")
+    parser.add_argument("bay", help="Laser bay to program for (2 or 3)")
+
+    args = parser.parse_args()
+
+    base_rate = int(args.base_rate)
+    goose_rate = int(args.goose_rate)
+    offset = int(args.offset)
+
+    allowed_bays = [2, 3]
+    if int(args.bay) not in allowed_bays:
+        raise ValueError(f"Bay {args.bay} not in {allowed_bays}!")
+    else:
+        bay = int(args.bay)
+
+    engines = {2: 6, 3: 7}  # Bay --> sequence engine mapping
+
+    # Dict will eventually be applied to drop down menu
+    base_list = make_base_rates(carbide_factors)
+
+    if base_rate not in base_list:
+        raise ValueError(
+           ("Base rate {base_rate} is not one of the available laser "
+            f"rates: {base_list}")
+        )
+
+    # Dict will eventually be applied to drop down menu
+    goose_list = allowed_goose_rates(base_rate, base_list)
+
+    if goose_rate not in goose_list:
+        raise ValueError(
+           (f"Goose rate {goose_rate} is not one of the available goose "
+            f"rates: {goose_list}")
+        )
+
+    seqdesc = {0: f"Bay {bay} On Time", 1: f"Bay {bay} Off Time", 2: "", 3: ""}
+    base_div = 910000//int(base_rate)
+    goose_div = 910000//int(goose_rate)
+    inst = make_sequence(base_div, goose_div, offset, True)
+
+    xpm_pv = "DAQ:NEH:XPM:0"
+    seqcodes_pv = Pv(f'{xpm_pv}:SEQCODES', isStruct=True)
+    seqcodes = seqcodes_pv.get()
+    desc = seqcodes.value.Description
+
+    engine = int(engines[bay])
+    seq = SeqUser(f'{xpm_pv}:SEQENG:{engine}')
+    seq.execute('title', inst, None, sync=True, refresh=False)
+
+    engineMask = 0
+    engineMask |= (1 << engine)
+
+    for e in range(4*engine, 4*engine+4):
+        desc[e] = ''
+    for e, d in seqdesc.items():
+        desc[4*engine+e] = d
+
+    tmo = 5.0  # epics pva timeout
```

Comment:
Didn't TMO want a faster timeout? Or am I thinking of the wrong sequencer?

Reply:
I'm not sure which sequencer you're talking about. If we reach the timeout of 5 seconds, something has gone wrong.

Reply:
It's only 4.5 mil times slower than the base rate 💀